src/sync.d

```diff
@@ -4469,7 +4474,16 @@ final class SyncEngine
 								if (response.type() == JSONType.object){
 									// upload done without error
 									writeln("done.");
-									// Log action to log file
+									
+									// upload finished
+									auto uploadFinishTime = Clock.currTime();
+									auto uploadDuration = uploadFinishTime - uploadStartTime;
+									log.vdebug("File Size: ", thisFileSize, " Bytes");
+									log.vdebug("Upload Duration: ", (uploadDuration.total!"msecs"/1e3), " Seconds");
+									auto uploadSpeed = (thisFileSize / (uploadDuration.total!"msecs"/1e3)/ 1024 / 1024);
```

Comment:
Can it happen that out of whatever reason `uploadDuration` is `0` (and the division breaks and throws exception)?

Reply:
@norbusan 
Totally plausible, but, based on start / finish times, the time breaks down to msec then usec and even one further time fraction.

Unless the code is being executed so quick that not even these values are being incremented, then yes, an exception will be thrown. If that happens however, someone can send me that PC hardware!

Reply:
Hahaha, good point! I also want such a hardware ;-)